### PR TITLE
KOGITO-5663 Improve path handling when Gradle runs the Quarkus extension

### PR DESCRIPTION
**Pull request by evacchi**
When the Quarkus extension runs through Gradle, resources cannot be detected because the paths assume a Maven directory layout. The task is to improve path handling so that resources can be found and built

I am adding a GRADLE, MAVEN enum so that a different prefix can be picked. I have currently tested this manually with Gradle using a BPMN file, but I can't give further assurance for DRL/DMN/PMML so far.

DMN is currently blocked by another classpath-related issue that I will investigate with the Quarkus team; see also https://issues.redhat.com/browse/KOGITO-5665

EDIT I am adding the exclusion to the BOM for the DMN issue, but that needs a fix on Quarkus side as well https://github.com/quarkusio/quarkus/pull/19307.

I am adding a gradle "integration test" by adding an almost empty module with a configured `exec` plugin. In the subdirectory of such module I am adding a Gradle project in a subdirectory, and running through the `exec` invocation.

~~The main issue is that the configuration parameters (version numbers) are hardcoded in `gradle.properties` and it would be better to read those from Maven properties. Maybe some `src/main/resources` filtering trickery could work?~~

Configuration parameters are read from Maven properties, and `gradle.properties` is generated from those, so this should be fine.

The test case currently is only making sure that the example builds successfully, we can improve this in the future.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Run all builds</b>  
  Please add comment: <b>Jenkins retest this</b>

* <b>Run (or rerun) specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [runtimes|optaplanner|apps|examples] tests</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Run (or rerun) LTS specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [runtimes|optaplanner|apps|examples] LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

* <b>Run (or rerun) native specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [runtimes|optaplanner|apps|examples] native</b>

* <b>Full Kogito testing</b> (with cloud images and operator BDD testing)  
  Please add comment: <b>Jenkins run BDD</b>  
  <b>This check should be used only if a big change is done as it takes time to run, need resources and one full BDD tests check can be done at a time ...</b>
</details>

### CODE REVIEW

**Comment by evacchi** [comment]
I have narrowed down this issue to `kogito-quarkus-decisions-extension`
the problem is that somehow `org.drools:drools-core-dynamic` is being pulled although there is an explicit exclusion in the dependency chain

For some reason `kogito-quarkus-rules-extension` is not affected!

workaround for now, add this to build.gradle :

```
configurations.all {
    exclude group: 'org.drools', module:'drools-core-dynamic'
}
```

**Comment by evacchi** [comment]
The other issue is that I am not 100% sure how we can test this. I am not a Gradle expert and we need to either

(1) figure out how to import the kogito BOM on a mismatching Quarkus platform 
(2) generate an ad-hoc Quarkus Platform 999-SNAPSHOT for the `main` branch with Kogito 2.0.0-SNAPSHOT; to be updated to the "current" (Quarkus) release on `stable` when Quarkus platform is released

File: integration-tests/integration-tests-quarkus-gradle/integration-tests-quarkus-gradle-project/gradle.properties [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#Gradle properties
+quarkusPluginId=io.quarkus
+quarkusPluginVersion=2.1.1.Final
+quarkusPlatformGroupId=io.quarkus.platform
+quarkusPlatformArtifactId=quarkus-bom
+quarkusPlatformVersion=2.1.1.Final
+kogitoVersion=2.0.0-SNAPSHOT
```
Comment (evacchi) [inline integration-tests/integration-tests-quarkus-gradle/integration-tests-quarkus-gradle-project/gradle.properties:7]: maybe we can expand Maven properties instead by using src/main/resources trickery?

Comment (evacchi) [reply integration-tests/integration-tests-quarkus-gradle/integration-tests-quarkus-gradle-project/gradle.properties]: added resources plugin: this file is now generated using maven properties!

File: integration-tests/integration-tests-quarkus-gradle/src/main/resources/gradle.properties [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Gradle properties
+quarkusPluginId=io.quarkus
+quarkusPluginVersion=2.1.1.Final
```
Comment (danielezonca) [inline integration-tests/integration-tests-quarkus-gradle/src/main/resources/gradle.properties:3]: Is it possible to use `${version.io.quarkus}` also here?
```suggestion
quarkusPluginVersion=${version.io.quarkus}
```

File: kogito-bom/pom.xml [hunk 3]
```diff
@@ -1430,6 +1430,17 @@
...
+        <exclusions>
+          <!--
+            Required for Gradle + Quarkus compat. See:
+            https://github.com/quarkusio/quarkus/pull/19307
+            https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/Gradle.20Kogito.20debugging
+          -->
+          <exclusion>
+            <groupId>org.drools</groupId>
+            <artifactId>drools-core-dynamic</artifactId>
+          </exclusion>
```
Comment (danielezonca) [inline kogito-bom/pom.xml:1442]: Can you create a ticket to track this issue? 
It is not clear to me if after the Quarkus fix ( https://github.com/quarkusio/quarkus/pull/19307 ) is released, we expect to remove this or if it is an additional/different problem

File: kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/utils/AppPaths.java [hunk 4]
```diff
@@ -88,10 +94,15 @@ private static PathType getPathType(Path archiveLocation) {
...
-    private AppPaths(Set<Path> projectPaths, Collection<Path> classesPaths, boolean isJar) {
+    private AppPaths(Set<Path> projectPaths, Collection<Path> classesPaths, boolean isJar, BuildTool bt) {
         this.isJar = isJar;
         this.projectPaths.addAll(projectPaths);
         this.classesPaths.addAll(classesPaths);
+        if (bt == BuildTool.GRADLE) {
+            resourcesPath = ""; // no prefix required
+        } else {
+            resourcesPath = "src/main/resources";
+        }
```
Comment (danielezonca) [inline kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/utils/AppPaths.java:105]: This code needs to be refactored to properly centralise and handle build tool discovery (similar to [this Quarkus class](https://github.com/quarkusio/quarkus/blob/main/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/utils/BuildToolHelper.java)).
Do you want to do it now or do you prefer to delegate to a different ticket (if ticket, please create it :) )?

File: quarkus/extensions/kogito-quarkus-extension-common/kogito-quarkus-common-deployment/src/main/java/org/kie/kogito/quarkus/common/deployment/KogitoQuarkusResourceUtils.java [hunk 5]
```diff
@@ -76,7 +76,13 @@ private KogitoQuarkusResourceUtils() {
 
     public static KogitoBuildContext kogitoBuildContext(Iterable<Path> paths, IndexView index, AppArtifact appArtifact) {
         // scan and parse paths
-        AppPaths appPaths = AppPaths.fromQuarkus(paths);
+        AppPaths.BuildTool buildTool;
+        if (System.getProperty("org.gradle.appname") == null) {
+            buildTool = AppPaths.BuildTool.MAVEN;
+        } else {
+            buildTool = AppPaths.BuildTool.GRADLE;
+        }
```
Comment (danielezonca) [inline quarkus/extensions/kogito-quarkus-extension-common/kogito-quarkus-common-deployment/src/main/java/org/kie/kogito/quarkus/common/deployment/KogitoQuarkusResourceUtils.java:84]: Same comment here :)

File: kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/utils/AppPaths.java [hunk 6]
```diff
@@ -26,18 +26,24 @@
...
+
     public static final String TARGET_DIR = "target";
 
     private final Set<Path> projectPaths = new LinkedHashSet<>();
     private final Collection<Path> classesPaths = new ArrayList<>();
 
     private final boolean isJar;
+    private final String resourcesPath;
 
     public static AppPaths fromProjectDir(Path projectDir) {
```
Comment (danielezonca) [inline kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/utils/AppPaths.java:42]: I think in the future we might consider to support `Gradle` also via `kogito-maven-plugin` (so also for Springboot). 
Wdyt? What about a ticket for that?

**Review by danielezonca — COMMENTED** [review]
Thanks for the PR :)

File: integration-tests/integration-tests-quarkus-gradle/src/main/resources/gradle.properties (hunk 2 above)
Comment (evacchi) [reply integration-tests/integration-tests-quarkus-gradle/src/main/resources/gradle.properties:3]: right! :)

File: kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/utils/AppPaths.java (hunk 4 above)
Comment (evacchi) [reply kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/utils/AppPaths.java:105]: I'll create a task and link it to [KOGITO-5456 Gradle support in Quarkus extension](https://issues.redhat.com/browse/KOGITO-5456)

File: kogito-bom/pom.xml (hunk 3 above)
Comment (evacchi) [reply kogito-bom/pom.xml:1442]: I think that's supposed to stay there.
@aloubyansky I assume the exclusion in the BOM is required right?

File: kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/utils/AppPaths.java [hunk 7]
```diff
@@ -88,10 +94,15 @@ private static PathType getPathType(Path archiveLocation) {
...
 
-    private AppPaths(Set<Path> projectPaths, Collection<Path> classesPaths, boolean isJar) {
+    private AppPaths(Set<Path> projectPaths, Collection<Path> classesPaths, boolean isJar, BuildTool bt) {
         this.isJar = isJar;
         this.projectPaths.addAll(projectPaths);
         this.classesPaths.addAll(classesPaths);
+        if (bt == BuildTool.GRADLE) {
+            resourcesPath = ""; // no prefix required
+        } else {
+            resourcesPath = "src/main/resources";
```
Comment (fjtirado) [inline kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/utils/AppPaths.java:104]: In order to make sure this is OS independent, Can we make resourcePath a Path object rather than a String and assing it to Path.get("src","main","resources")
I know this imply more changes, but it is worth to avoid having issues in windows (which are hard to debug for us)

File: kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/utils/AppPaths.java (hunk 6 above)
Comment (aloubyansky) [reply kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/utils/AppPaths.java:42]: I think the right approach would be switch to (evolving if necessary) the Quarkus code-gen mechanism that should work in both Maven and Gradle projects.

File: kogito-bom/pom.xml (hunk 3 above)
Comment (aloubyansky) [reply kogito-bom/pom.xml:1442]: Yes, this should stay even after the Quarkus issue has been fixed.

File: kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/utils/AppPaths.java (hunk 4 above)
Comment (evacchi) [reply kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/utils/AppPaths.java:105]: https://issues.redhat.com/browse/KOGITO-5695

File: kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/utils/AppPaths.java (hunk 6 above)
Comment (evacchi) [reply kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/utils/AppPaths.java:42]: +1 @aloubyansky -- however we are also shipping a "pure" Maven plug-in with a different codegen backend for Spring users; @danielezonca was suggesting to cater for those Spring users by possibly providing a Gradle plugin as well.

File: integration-tests/integration-tests-quarkus-gradle/integration-tests-quarkus-gradle-project/build.gradle [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    testImplementation 'io.rest-assured:rest-assured'
+}
+
+group 'org.kie.kogito.examples'
+version '1.0.0-SNAPSHOT'
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_11
+    targetCompatibility = JavaVersion.VERSION_11
+}
```
[the remark this answers is not in the document]
Comment (evacchi) [reply integration-tests/integration-tests-quarkus-gradle/integration-tests-quarkus-gradle-project/build.gradle:28]: mm not sure because syntactically those look like enums, I don't know if they are actually Strings.

File: integration-tests/integration-tests-quarkus-gradle/integration-tests-quarkus-gradle-project/build.gradle [hunk 9]
```diff
@@ -0,0 +1,37 @@
+plugins {
+    id 'java'
+    id 'io.quarkus'
+}
+
+repositories {
+    mavenCentral()
+    mavenLocal()
+}
```
[the remark this answers is not in the document]
Comment (evacchi) [reply integration-tests/integration-tests-quarkus-gradle/integration-tests-quarkus-gradle-project/build.gradle:9]: we may be able to do that if we can use the templating system somehow. I don't Gradle enough at the moment. I just wanted to at least have a rough test case in our repo, but we will work on this further next sprint (as planned)

File: integration-tests/integration-tests-quarkus-gradle/integration-tests-quarkus-gradle-project/build.gradle (hunk 8 above)
[the remark this answers is not in the document]
Comment (radtriste) [reply integration-tests/integration-tests-quarkus-gradle/integration-tests-quarkus-gradle-project/build.gradle:28]: oki

File: integration-tests/integration-tests-quarkus-gradle/integration-tests-quarkus-gradle-project/build.gradle (hunk 9 above)
[the remark this answers is not in the document]
Comment (radtriste) [reply integration-tests/integration-tests-quarkus-gradle/integration-tests-quarkus-gradle-project/build.gradle:9]: I am checking that the set version does not impact the tests

[the remark this answers is not in the document]
Comment (radtriste) [reply integration-tests/integration-tests-quarkus-gradle/integration-tests-quarkus-gradle-project/build.gradle:9]:
```
A problem occurred configuring root project 'integration-tests-quarkus-gradle-project'.
> Could not resolve all dependencies for configuration ':quarkusDependency'.
   > Could not find org.kie.kogito:kogito-bom:754.0.0.Final.
     Searched in the following locations:
       - https://repo.maven.apache.org/maven2/org/kie/kogito/kogito-bom/754.0.0.Final/kogito-bom-754.0.0.Final.pom
       - file:/home/tradisso/.m2/repository/org/kie/kogito/kogito-bom/754.0.0.Final/kogito-bom-754.0.0.Final.pom
     Required by:
         project :
   > Could not resolve org.kie.kogito:kogito-quarkus-processes:754.0.0.Final.
     Required by:
         project :
      > Could not resolve org.kie.kogito:kogito-quarkus-processes:754.0.0.Final.
         > Could not parse POM /home/tradisso/.m2/repository/org/kie/kogito/kogito-quarkus-processes/754.0.0.Final/kogito-quarkus-processes-754.0.0.Final.pom
            > Could not resolve org.kie.kogito:kogito-quarkus-processes-extension:754.0.0.Final.
               > Could not resolve org.kie.kogito:kogito-quarkus-processes-extension:754.0.0.Final.
                  > Could not parse POM /home/tradisso/.m2/repository/org/kie/kogito/kogito-quarkus-processes-extension/754.0.0.Final/kogito-quarkus-processes-extension-754.0.0.Final.pom
                     > Could not resolve org.kie.kogito:quarkus-extensions:754.0.0.Final.
                        > Could not resolve org.kie.kogito:quarkus-extensions:754.0.0.Final.
                           > Could not parse POM /home/tradisso/.m2/repository/org/kie/kogito/quarkus-extensions/754.0.0.Final/quarkus-extensions-754.0.0.Final.pom
                              > Could not resolve org.kie.kogito:quarkus:754.0.0.Final.
                                 > Could not resolve org.kie.kogito:quarkus:754.0.0.Final.
                                    > Could not parse POM /home/tradisso/.m2/repository/org/kie/kogito/quarkus/754.0.0.Final/quarkus-754.0.0.Final.pom
                                       > Could not resolve org.kie.kogito:kogito-build-parent:754.0.0.Final.
                                          > Could not resolve org.kie.kogito:kogito-build-parent:754.0.0.Final.
                                             > Could not parse POM /home/tradisso/.m2/repository/org/kie/kogito/kogito-build-parent/754.0.0.Final/kogito-build-parent-754.0.0.Final.pom
                                                > Could not find org.kie.kogito:kogito-bom:754.0.0.Final.

```
I will check if this can be resolved quickly ...

[the remark this answers is not in the document]
Comment (radtriste) [reply integration-tests/integration-tests-quarkus-gradle/integration-tests-quarkus-gradle-project/build.gradle:9]: or that would mean that kogito-bom is not handled before `integration-tests-quarkus-gradle` ?
that would mean some things may also fail on SNAPSHOT ones if modifications are done ...

Command that I ran: `$ mvn clean install -pl :integration-tests-quarkus-gradle -am`

[the remark this answers is not in the document]
Comment (radtriste) [reply integration-tests/integration-tests-quarkus-gradle/integration-tests-quarkus-gradle-project/build.gradle:9]:
```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Kogito Runtimes 754.0.0.Final:
[INFO] 
[INFO] Kogito Runtimes .................................... SUCCESS [  0.630 s]
[INFO] Kogito :: IDE/Tools - Config ....................... SUCCESS [  0.453 s]
[INFO] Kogito :: Runtimes Build Parent .................... SUCCESS [  1.257 s]
[INFO] Kogito :: API ...................................... SUCCESS [  0.059 s]
[INFO] Kogito :: API :: Events ............................ SUCCESS [  3.445 s]
[INFO] Kogito :: API :: Timer ............................. SUCCESS [  1.090 s]
[INFO] Kogito :: Drools ................................... SUCCESS [  0.034 s]
[INFO] Kogito :: Drools :: Kogito Predictive Model Markup Language :: API Dependencies SUCCESS [  0.116 s]
[INFO] Kogito :: API :: Public API ........................ SUCCESS [  2.790 s]
[INFO] Kogito :: API :: Services .......................... SUCCESS [  2.342 s]
[INFO] Kogito :: Drools :: Drools Core Static ............. SUCCESS [  1.200 s]
[INFO] Kogito :: Drools :: Drools Core .................... SUCCESS [  0.855 s]
[INFO] Kogito :: Drools :: Drools Compiler ................ SUCCESS [  0.286 s]
[INFO] Kogito :: Drools :: Kogito Drools Model ............ SUCCESS [  0.335 s]
[INFO] Kogito :: Drools :: Kogito RuleUnits ............... SUCCESS [  1.697 s]
[INFO] Kogito :: Drools :: Kogito Drools .................. SUCCESS [  0.085 s]
[INFO] Kogito :: Drools :: Kogito Decision Model Notation . SUCCESS [  1.751 s]
[INFO] Kogito :: Drools :: Drools Core Dynamic ............ SUCCESS [  0.072 s]
[INFO] Kogito :: Drools :: Kogito Predictive Model Markup Language :: Dependencies SUCCESS [  0.169 s]
[INFO] Kogito :: Drools :: Kogito Predictive Model Markup Language :: OpenAPI SUCCESS [  1.578 s]
[INFO] Kogito :: Drools :: Kogito Predictive Model Markup Language SUCCESS [  1.591 s]
[INFO] Kogito :: Drools :: Drools DecisionTables .......... SUCCESS [  0.100 s]
[INFO] Kogito :: jBPM ..................................... SUCCESS [  0.029 s]
[INFO] Kogito :: Process :: WorkItems ..................... SUCCESS [  0.384 s]
[INFO] Kogito :: jBPM :: Flow ............................. SUCCESS [  9.217 s]
[INFO] Kogito :: Process :: Protobuf Serialization ........ SUCCESS [  5.074 s]
[INFO] Kogito :: jBPM :: Flow Builder ..................... SUCCESS [ 22.416 s]
[INFO] Kogito :: jBPM :: BPMN2 ............................ SUCCESS [02:09 min]
[INFO] Kogito :: Codegen Modules .......................... SUCCESS [  0.027 s]
[INFO] Kogito :: Codegen API .............................. SUCCESS [  1.866 s]
[INFO] Kogito :: Codegen Core Library ..................... SUCCESS [  2.205 s]
[INFO] kogito-serverless-workflow ......................... SUCCESS [  0.027 s]
[INFO] Kogito Serverless Workflow Runtime ................. SUCCESS [  0.153 s]
[INFO] Kogito :: Workitems ................................ SUCCESS [  0.027 s]
[INFO] Kogito :: Workitems :: RestWorkItem ................ SUCCESS [  2.210 s]
[INFO] Kogito :: Workitems :: OpenApiWorkItem ............. SUCCESS [  1.529 s]
[INFO] Kogito :: Serverless Workflow Builder .............. SUCCESS [  2.734 s]
[INFO] Kogito :: Codegen OpenApi .......................... SUCCESS [  5.321 s]
[INFO] Kogito :: Codegen Processes ........................ SUCCESS [  7.958 s]
[INFO] Kogito :: Grafana API .............................. SUCCESS [  1.654 s]
[INFO] Kogito :: Codegen Rules ............................ SUCCESS [  4.830 s]
[INFO] Kogito :: Codegen Decisions ........................ SUCCESS [  5.968 s]
[INFO] Kogito :: Codegen Predictions ...................... SUCCESS [  5.346 s]
[INFO] Kogito :: Quarkus :: Parent ........................ SUCCESS [  0.022 s]
[INFO] Kogito :: Quarkus Extensions :: Parent ............. SUCCESS [  0.024 s]
[INFO] Kogito :: Quarkus Extension Common ................. SUCCESS [  0.034 s]
[INFO] Kogito :: Quarkus Extension Common :: Runtime ...... SUCCESS [  0.477 s]
[INFO] Kogito :: Quarkus Extension Common :: Deployment ... SUCCESS [  0.710 s]
[INFO] Kogito :: Quarkus Rules Extension .................. SUCCESS [  0.025 s]
[INFO] Kogito :: Quarkus Rules Extension :: Runtime ....... SUCCESS [  0.467 s]
[INFO] Kogito :: Quarkus Decisions Extension .............. SUCCESS [  0.028 s]
[INFO] Kogito :: Quarkus Decisions Extension :: Runtime ... SUCCESS [  0.193 s]
[INFO] Kogito :: Quarkus Predictions Extension ............ SUCCESS [  0.025 s]
[INFO] Kogito :: Quarkus Predictions Extension :: Runtime . SUCCESS [  0.176 s]
[INFO] Kogito :: Quarkus Processes Extension .............. SUCCESS [  0.025 s]
[INFO] Kogito :: Quarkus Processes Extension :: Runtime ... SUCCESS [  0.173 s]
[INFO] Kogito :: Quarkus Extension ........................ SUCCESS [  0.031 s]
[INFO] Kogito :: Quarkus Extension :: Runtime ............. SUCCESS [  0.186 s]
[INFO] Kogito :: Quarkus Rules Extension :: Deployment .... SUCCESS [  0.387 s]
[INFO] Kogito :: Quarkus Decisions Extension :: Deployment  SUCCESS [  0.333 s]
[INFO] Kogito :: Quarkus Predictions Extension :: Deployment SUCCESS [  0.325 s]
[INFO] Kogito :: Quarkus Processes Extension :: Deployment  SUCCESS [  1.907 s]
[INFO] Kogito :: Quarkus Extension :: Deployment .......... SUCCESS [  0.072 s]
[INFO] Kogito :: Integration Tests ........................ SUCCESS [  0.019 s]
[INFO] Kogito :: Integration Tests :: Quarkus :: Gradle ... FAILURE [ 33.977 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  04:31 min
[INFO] Finished at: 2021-08-11T16:12:52+02:00
[INFO] ------------------------------------------------------------------------

```

[the remark this answers is not in the document]
Comment (radtriste) [reply integration-tests/integration-tests-quarkus-gradle/integration-tests-quarkus-gradle-project/build.gradle:9]: cannot reproduce the problem with a full `mvn clean install`  so I guess we are good with this :)

**Review by radtriste — APPROVED** [review]

**Comment by evacchi** [comment]
as detailed on Zulip, the Gradle version used there is whatever version is being shipped with `quarkus create app --gradle my-app`; OTOH, your particular issue may be addressed by simply skipping the build -- the global property is currently not being honored on `main`; see https://github.com/kiegroup/kogito-runtimes/pull/1535
